sql/migrations/007_user_groups.sql

```diff
@@ -0,0 +1,20 @@
+-- +migrate Up
+
+ALTER TABLE "task_template" ADD COLUMN "allowed_resolver_groups" JSONB NOT NULL DEFAULT '[]';
+ALTER TABLE "task" ADD COLUMN "resolver_groups" JSONB NOT NULL DEFAULT 'null';
+ALTER TABLE "task" ADD COLUMN "watcher_groups" JSONB NOT NULL DEFAULT 'null';
+ALTER TABLE "task" ADD COLUMN "requester_groups" JSONB NOT NULL DEFAULT 'null';
+
+CREATE INDEX "task_resolver_groups_idx" ON "task" USING gin (resolver_groups jsonb_path_ops);
+CREATE INDEX "task_watcher_groups_idx" ON "task" USING gin (watcher_groups jsonb_path_ops);
```

Comment:
```suggestion
CREATE INDEX "task_resolver_groups_idx" ON "task" USING gin (resolver_groups);
CREATE INDEX "task_watcher_groups_idx" ON "task" USING gin (watcher_groups);
```

According to Postgres documentation, https://www.postgresql.org/docs/14/datatype-json.html#JSON-INDEXING, it seems that operator `?|` is not supported by `jsonb_path_ops`, so I suggest that we use the native `jsonb` index class.